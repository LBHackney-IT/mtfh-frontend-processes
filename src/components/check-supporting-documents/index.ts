export * from "./check-supporting-documents";
