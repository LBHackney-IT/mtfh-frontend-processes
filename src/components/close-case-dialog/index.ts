export * from "./close-case-dialog";
