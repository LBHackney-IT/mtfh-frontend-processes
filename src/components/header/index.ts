export * from "./sole-to-joint-header";
