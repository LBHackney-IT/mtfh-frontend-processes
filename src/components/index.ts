export * from "./start-process";
export * from "./entity-summary";
export * from "./process-menu";
