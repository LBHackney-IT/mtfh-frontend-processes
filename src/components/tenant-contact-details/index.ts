export * from "./tenant-contact-details";
