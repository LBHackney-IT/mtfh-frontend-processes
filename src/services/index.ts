export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
