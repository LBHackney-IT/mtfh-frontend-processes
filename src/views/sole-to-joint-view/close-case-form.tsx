import React from "react";

import * as Yup from "yup";

import { locale } from "../../services";

import { Field, TextArea } from "@mtfh/common/lib/components";

const { reasonForCancellation, reasonForCloseCase } = locale.views.closeCase;

export const closeCaseSchema = (errorMessages: Record<string, string>) =>
  Yup.object({
    reasonForCancellation: Yup.string().required(errorMessages.W50),
  });

export type CloseCaseFormData = Yup.Asserts<ReturnType<typeof closeCaseSchema>>;

export const CloseCaseForm = ({ isCancel }): JSX.Element => {
  return (
    <>
      <Field
        id="mtfh-close-case-form-reason"
        name="reasonForCancellation"
        label={isCancel ? reasonForCancellation : reasonForCloseCase}
        required
      >
        <TextArea rows={5} />
      </Field>
    </>
  );
};
