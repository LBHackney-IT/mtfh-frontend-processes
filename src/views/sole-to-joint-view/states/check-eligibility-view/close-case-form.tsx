import React from "react";

import * as Yup from "yup";

import { locale } from "../../../../services";

import { Field, TextArea } from "@mtfh/common/lib/components";

const { reasonForRejection } = locale.views.closeCase;

export const closeCaseSchema = (errorMessages: Record<string, string>) =>
  Yup.object({
    reasonForRejection: Yup.string().required(errorMessages.W50),
  });

export type CloseCaseFormData = Yup.Asserts<ReturnType<typeof closeCaseSchema>>;

export const CloseCaseForm = (): JSX.Element => {
  return (
    <>
      <Field
        id="mtfh-close-case-form-reason"
        name="reasonForRejection"
        label={reasonForRejection}
        required
      >
        <TextArea rows={5} />
      </Field>
    </>
  );
};
