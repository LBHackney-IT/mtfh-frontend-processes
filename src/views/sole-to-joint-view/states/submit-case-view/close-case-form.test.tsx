import React, { useState } from "react";

import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Form, Formik } from "formik";

import { locale } from "../../../../services";
import { CloseCaseForm, CloseCaseFormData, closeCaseSchema } from "./close-case-form";

import { Button } from "@mtfh/common/lib/components";
import commonLocale from "@mtfh/common/lib/locale";

const mockErrorMessages = commonLocale.hooks.defaultErrorMessages;

const Component = ({
  initialValues,
}: {
  initialValues?: Partial<CloseCaseFormData>;
}): JSX.Element => {
  const [submitted, setSubmitted] = useState(false);

  if (submitted) {
    return <div>Success</div>;
  }

  return (
    <Formik
      initialValues={{
        reasonForRejection: "",
        ...initialValues,
      }}
      validationSchema={closeCaseSchema(mockErrorMessages)}
      validateOnBlur={false}
      validateOnChange={false}
      onSubmit={() => {
        setSubmitted(true);
      }}
    >
      {() => (
        <Form noValidate data-testid="form">
          <CloseCaseForm />
          <Button type="submit">{locale.confirm}</Button>
        </Form>
      )}
    </Formik>
  );
};

test("it renders correctly", () => {
  const { container } = render(<Component />);
  expect(container).toMatchSnapshot();
});

test("it fails validation if submitted without interaction", async () => {
  render(<Component />);
  await userEvent.click(screen.getByText(locale.confirm));
  await screen.findByText("Error", { exact: false });
});

test("it submits with correct required fields", async () => {
  render(<Component />);
  const reasonForRejection = screen.getByLabelText(
    `${locale.views.closeCase.reasonForRejection}*`,
  );
  await userEvent.type(reasonForRejection, "Documents not provided");
  await userEvent.click(screen.getByText(locale.confirm));
  await screen.findByText("Success");
});
