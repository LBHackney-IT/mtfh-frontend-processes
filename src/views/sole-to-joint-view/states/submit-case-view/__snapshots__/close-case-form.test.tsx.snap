// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <form
    action="#"
    data-testid="form"
    novalidate=""
  >
    <div
      class="govuk-character-count"
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="mtfh-close-case-form-reason"
      >
        <label
          class="govuk-label lbh-label"
          for="mtfh-close-case-form-reason-field"
        >
          Reason for Rejection
          <sup
            aria-hidden="true"
          >
            *
          </sup>
        </label>
        <textarea
          class="govuk-textarea lbh-textarea lbh-character-count"
          id="mtfh-close-case-form-reason-field"
          name="reasonForRejection"
          required=""
          rows="5"
        />
      </div>
    </div>
    <button
      class="govuk-button lbh-button"
      type="submit"
    >
      Confirm
    </button>
  </form>
</div>
`;
